### ISSUE TITLE
Can't use fun<RetVal, MyClass>(&MyClass::member_func) if the member function is const in v5.8.6

### ISSUE THREAD

**Issue by tankorsmash** [ChaiScript/ChaiScript#526]
* Compiler Used: VS2013
 * Operating System: Windows 10
 * Architecture (ARM/x86/32bit/64bit/etc): 
 

### Expected Behavior 
Being able to use `chai.add(fun<RetVal, MyClass>(&MyClass::member_func), "member_func")`

### Actual Behavior 

I need to use `auto ptr = static_cast<RetVal (MyClass::*)() const>; chai.add(fun(ptr), "member_func"); to register the member function.

### Solution

I'm sure this is is solved in v6+, but if anyone is stuck on v5.8.6 like I am, just add a new overload. It appears to work:

```cpp
namespace chaiscript{
    template<typename Ret, typename Class>
    Proxy_Function fun(Ret(Class::*t_func)() const)
    {
        auto call = dispatch::detail::Const_Caller<Ret, Class>(t_func);

        return Proxy_Function(
            chaiscript::make_shared<dispatch::Proxy_Function_Base, dispatch::Proxy_Function_Callable_Impl<Ret(const Class &), decltype(call)>>(call));
    }
}
```

Basically just copied and pasted it from v5.8.6's register_function.hpp and stripped out the `Params...` part of the template signature.

**Comment by tankorsmash** [comment]
Closing this just so that someone else can find a solution to their issue without digging as much as I did.
